### PR TITLE
Add an edit version number to the header

### PR DESCRIPTION
**Pull request by robert-thompson**
- Add a version number to the file contents to track an edit version.

- If a file is copied instead of cloned then the version control
  information is lost after the copy. The edit version is not.

- While the edit version **may** not match the version control version they
  **should** match. Updating an option in the configuration **must**
  update the edit version. Updating spacing, comments, or anything that
  does not alter how the database operates **should** update the edit version.
  In other words, every commit **may** not need a version bump unless the
  change alters the running configuration.

### CODE REVIEW

**Comment by terackspace** [comment]
Reviewing this, the only change I don't like is the RHEL 5.1 terse file going from 2.12 -> 1.00. I feel this is a bad idea, as there are probably thousands of versions of this config out there -- can we just leave this version in that range, and bump it to 2.13? These configs are consumed outside of fresh installations and/or automation, folks have been manually deploying them from their older IUS git repo not related to specific IUS usage for years.

**Comment by robert-thompson** [comment]
Corrected the version to 2.13. My initial idea was to make the version relative to this repository but that's a good point.

I also cleaned up some trailing whitespaces.

**Comment by carlgeorge** [comment]
I definitely agree with @terackspace that the version number shouldn't move backwards.

Regarding the latest commits on this pull, here are the versions of the files before and after this proposed change.

file | before | after
---|---|---
my-5.1-rhel.cnf | v2.12 | v2.13
my-5.5-rhel.cnf | none | v1.00
my-5.5-ubuntu.cnf | none | v1.00
my-5.6-rhel.cnf | none | v1.00
 
_Note: The RHEL 5.5 and 5.6 config files up to this point have not had a version number.  That was how IUS received them from the DBA team._

Something about having completely different version in these files rubs me the wrong way.  In my mind, the version number is related to the DBA's recommended defaults, and the actual file is just the distro specific implementation of those recommended defaults.  I propose we establish a convention for these version numbers that allows for clarity and consistency between distros and MySQL releases.  It might look like this:

```
Given a version number MAJOR.MINOR.PATCH, increment the:

    MAJOR version when a major change occurs (such as a new versioning scheme).
    MINOR version when a change occurs that applies to all configurations.
    PATCH version when a change occurs that applies to a single configuration.
```

Practical example:

* RHEL and Ubuntu configs are at 3.0.0.
* We apply a change to the Ubuntu config that isn't necessary/relevant to the RHEL config, so we bump the Ubuntu config version to 3.0.1, but leave the RHEL config at 3.0.0.
* We apply a change that applies to all configs, so we bump all the config files to 3.1.0.

Hey @abg, what do you think?  We could use a DBA viewpoint on this.

**Comment by robert-thompson** [comment]
Added Semantic Version numbering, bumped the version 'higher' than the previous version (though changing the version scheme breaks compatibility), and made the new version consistent across all files.

I did this in a second commit to preserve the old version number in the history.

**Comment by carlgeorge** [comment]
Thanks @robert-thompson.  I'm fine with merging this, but since it was my suggestion, I'll give other interested parties a chance to disagree first.

**Comment by abg** [comment]
I had the discussion about version numbers with B.J. Dierkes many years ago.  The version numbers were pretty meaningless back in 2008 when I started contributing to them and around the 5.1 lifecycle we dropped them altogether from the configs. I don't recall why 5.1 wasn't updated - we may have been avoiding any serious change to the packages at the time.

Honestly the versions embedded in the configs themselves aren't particularly useful - the configs are going to be significantly modified in any serious production environment (in many cases manually).  These are just a proposed set of defaults to workaround issues we've seen with the upstream defaults in Rackspace customer environments in the past.

With that said, I think the proposed versioning is pretty harmless and if it is maintained and others find it useful I have no objections.

**Comment by robert-thompson** [comment]
Yeah, I agree that the version number becomes less useful for version tracking once the files are used but what I am trying to answer is what version is in this repo versus what version is in ADC. The authoritative way is to checksum the file but that information could also be gleaned from a version number (with the assumption that the version numbers in the files are being updated).

With that in mind I redid the pull request with:

1. Removed the version number from the top
2. Moved it to the bottom of cowsay with the following: "Based on vX.X.X"
3. Added a link to this repo to complete the circle.
4. Kept the version number 3.0.0 to keep any resemblance of backwards compatibility with previous versions.

I believe this will allow me to know if a base file I'm using for ADC has changed but not imply that a version on a server == a version here.

**Comment by carlgeorge** [comment]
:thumbsup:

**Comment by tonyskapunk** [comment]
Personally don't think there is great value on adding a version on the config file itself, unless, it is being maintained, if this is the plan with this repo then :+1:  @carlgeorge do you want to do the merge or are you OK if I do this?  Do we want to have any specific number of +1 before that?

**Comment by terackspace** [comment]
@tonyskapunk - files "travel", I think maintaining versions embedded in them has intrinsic value; it allows for a user who *may have* kept downloaded files locally (like, say, me :) ) to easily tell if the version they have is the newest or not before copying it up to a server. Files traveling around without version numbers lead to confusion and such, so depending on the use case can be valuable.

We also should recognize that we rarely update these files, so the maintenance cost of bumping a version number in a file manually before the commit is very, very low (I'm talking once a year if that). I give a +1 to @carlgeorge merging the latest revision of the PR, I think we're good.

**Comment by carlgeorge** [comment]
Merged and added tag for 3.0.0.
